### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by sun0225SUN** [lowlighter/metrics#1344] [Q&A] [answered]
sorry，I want to generate multiple charts and store them in one folder, what should I do？

![image](https://user-images.githubusercontent.com/79169717/211562872-170dcd25-aaf6-4186-89bf-0f121377c721.png)
![image](https://user-images.githubusercontent.com/79169717/211563488-06df1224-197e-40e2-9ba2-598b5e947f39.png)

**Comment by sun0225SUN** [comment, marked as answer]
![2023-01-10_23-25-50](https://user-images.githubusercontent.com/79169717/211592142-18b3d911-47d4-43ac-88cc-2148ad31f63c.png)

ok，I solved it, I created a new branch, and let the chart be generated in this branch, so that the main branch looks better, hahaha！

**Reply by lowlighter** [reply]
Glad you found a way to do it ! 

Note that despite the name `filename` should also support directories (relative to repository root) so `directory/metrics.svg` would also work. But I think it's cleaner to have a separate branch too

**Reply by sun0225SUN** [reply]
I think I found the solution, but after this I've been researching how GitHub actions work. 
For example how to automatically trigger workflows on non-main branches? 
How to set workflows under the main branch, but output the target files to other branches?

Now I know that the file name supports setting the directory, like `directory/metrics.svg`, I decided to put all the workflows under the main branch,

thank you very much, you built an excellent project, wish you have a good life!

By the way, my profile is more beautiful because of your project, haha, thank you!

**Reply by lowlighter** [reply]
> For example how to automatically trigger workflows on non-main branches?

I suppose [`schedule`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) trigger should work on non main branches, though I never personally tested:

The `examples.yml` workflow on this repository uses a scheduler on the main branche but actually switchout to another branch so maybe that may help in what you'd like to achieve in the future

https://github.com/lowlighter/metrics/blob/cec8d889ac8f58c13b31f1de7642f991e290906b/.github/workflows/examples.yml#L2-L5
https://github.com/lowlighter/metrics/blob/cec8d889ac8f58c13b31f1de7642f991e290906b/.github/workflows/examples.yml#L43-L48
https://github.com/lowlighter/metrics/blob/cec8d889ac8f58c13b31f1de7642f991e290906b/.github/workflows/examples.yml#L1406-L1416

Thanks a lot for all your support and wish you the best too ! 
I'm glad you like this project and your profile does look awesome indeed !
